faceV4.py

```diff
@@ -0,0 +1,202 @@
+import requests
+from datetime import datetime, timedelta
+import time
+import json
+import calendar
+import csv
+import sys
+import os
+import argparse
+import urllib3
+
+#to disable warnings for not verifying host
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+def argparser():
+    my_parser = argparse.ArgumentParser()
+
+    #aruements avaiable for the user to customize
+    my_parser.add_argument('APIkey', type=str, help='Get this from your console')
+    my_parser.add_argument('-s', '--startTime', type=str, help='Add the end search time in yyyy-mm-dd~(0)0:00:00 or default to 1 hour before current time')
+    my_parser.add_argument('-e', '--endTime', type=str, help='Add the end search time in yyyy-mm-dd~(0)0:00:00 or default to current time')
+    my_parser.add_argument('-f', '--filter', type=str, help= 'Choose a filter', choices=['alert','trusted','named','other'], default='named')
+    my_parser.add_argument('-n', '--name', type = str, help = 'Searches for name')
+    my_parser.add_argument('-c', '--cameraName', type=str, help='Name of camera in the console')
+    my_parser.add_argument('--csv', type=str, help= 'Name the csv file', default='report')
+    my_parser.add_argument('-r', '--report', type=str, help='Name the folder for csv file and thumbnails', default='Report')
+
+    args = my_parser.parse_args()
+    return args
+
+args = argparser()
+
+#decrypts the jpg and saves the image to a folder
+def saving_img(thumbnail, name, count):
+    if __name__ == "__main__":
+        #url of the api
+        endpoint = thumbnail
+
+        api_key = args.APIkey
+
+        sess = requests.session()
```

Comment:
We only need one requests session for this example, since it holds all the required data for any of the api calls.  I would suggest making the example into a class and having the session be a class variable that you use in any of the calls.  The headers can be set at the session level and wouldn't need to be manually set everytime.